### PR TITLE
Server-side Pagination

### PR DESCRIPTION
**Pull request by emilyliublair**
## Server-side Pagination

Issue Number(s): #189 

What does this PR change and why?
implements serverside pagination

### Checklist

- [ ] Requirements have been implemented
- [ ] Acceptance criteria is met
- [ ] Database schema [docs](https://www.notion.so/gtbitsofgood/Database-Schema-Docs-fd377ded10b54890b65fd99025488cee?pvs=4) have been updated or are not necessary
- [ ] Relevant reviewers (EM) have been assigned to this PR

### Critical Changes

- Database change / migration to run
- Environment config change
- Breaking API change

### Related PRs

PR Number(s): #number_of_pr

### How To Test

check to see that only certain number of data is returned rather than all

### CODE REVIEW

**Comment by dograh** [comment]
Design looks good! Great work!

**Review by laurenmcp — COMMENTED** [review]
Looks awesome emily! 
One thing- there seems to be a client side error on the Access management page. Could you investigate this? I'm thinking the access management table may have a dependency on the table component, but Alexa will be able to provide more guidance on why this may be happening.

File: src/components/Table/Table.jsx [hunk 1]
```diff
@@ -40,11 +40,11 @@ function TableColumn({ icon, subLabel, col, style }) {
   );
 }
 
-function TableFooter({ elementsOnPage, rows, paginationFunctions }) {
+function TableFooter({ paginationFunctions }) {
   return (
     <div className="flex bg-white justify-between items-center px-6 py-4 w-[90vw] sm:w-full mt-2 sm:mt-0 rounded-lg sm:rounded-none border-gray-300 border-[1px] sm:border-0">
```
Comment (afazio1) [inline src/components/Table/Table.jsx]: change `w-[90vw]` to `w-full`

**Review by afazio1 — CHANGES_REQUESTED** [review]
Looks awesome!! 

A few things:
- When there are 0 logs / dogs, pagination should be clamped. But I am able to click it and get NaN
Screenshot 2024-04-09 at 4 47 43 PM

**Review by afazio1 — APPROVED** [review]

**Pull request merged**